oak_containers_hello_world_trusted_app/src/app_service.rs

```diff
@@ -44,10 +47,53 @@ impl TrustedApplication for TrustedApplicationImplementation {
         &self,
         request: tonic::Request<HelloRequest>,
     ) -> Result<tonic::Response<HelloResponse>, tonic::Status> {
-        let name = request.into_inner().name;
+        let encrypted_request = request
+            .into_inner()
+            .encrypted_request
+            .ok_or(tonic::Status::internal("encrypted request wasn't provided"))?;
+
+        // Initialize server encryptor.
+        let serialized_encapsulated_public_key = encrypted_request
+            .serialized_encapsulated_public_key
+            .as_ref()
+            .ok_or(tonic::Status::invalid_argument(
+                "initial request message doesn't contain encapsulated public key",
+            ))?;
+        let serialized_crypto_context = self
+            .orchestrator_client
+            .get_crypto_context(serialized_encapsulated_public_key)
+            .await
+            .map_err(|error| {
+                tonic::Status::internal(format!(
+                    "couldn't get crypto context from the Orchestrator: {:?}",
+                    error
+                ))
+            })?;
+        let crypto_context =
+            RecipientContext::deserialize(serialized_crypto_context).map_err(|error| {
+                tonic::Status::internal(format!("couldn't deserialize crypto context: {:?}", error))
+            })?;
+        let mut server_encryptor = ServerEncryptor::new(crypto_context);
+
+        // Associated data is ignored.
+        let (name_bytes, _) = server_encryptor
+            .decrypt(&encrypted_request)
+            .map_err(|error| {
+                tonic::Status::internal(format!("couldn't decrypt request: {:?}", error))
+            })?;
```

Comment:
The real application logic here is very obfuscated. The business logic is just the line with `format!` below, and the rest is boilerplate that is the same for all other apps. Even though we should always make it possible for an application to have access to the encryptor / decryptor objects, I imagine for most apps something more similar to how we do this for Oak Restricted Kernel may be enough:

https://github.com/project-oak/oak/blob/e4d09f685092aa1969d05762a779cf5e6692cb1e/oak_functions_service/src/lib.rs#L120-L132

Reply:
Creating an encryptor requires an `async` `get_crypto_context` function, so it couldn't be put into the `oak_crypto` library, which should also work in the RKernel use-case without `async`.

Reply:
Once we add `async` to the RKernel then https://github.com/project-oak/oak/pull/4281 could be reverted.

And we would have a `CryptoContextGenerator` trait, which will be implemented by the `OrchestratorClient` in Containers and by `KernelClient` in RKernel.

But for now yes, we potentially can have 2 separate handlers. Though the `async` implementation should not even be in the `oak_crypto` create, since it's an RKernel dependency.
Or we can have `async` functions in the dependency library even though RKernel doesn't support `async`?